python2verilog/frontend/generator2ir.py

```diff
@@ -176,14 +173,16 @@ def __parse_statement(self, stmt: pyast.AST, nextt: ir.Element, prefix: str):
                 nextt.unique_id,
             )
         elif isinstance(stmt, pyast.Yield):
-            cur_node = self.__parse_yield(stmt, prefix=prefix)
+            cur_node, end_node = self.__parse_yield(stmt, prefix=prefix)
             edge = ir.ClockedEdge(unique_id=f"{prefix}_e", child=nextt)
-            cur_node.child = edge
+            end_node.child = edge
         elif isinstance(stmt, pyast.While):
             cur_node = self.__parse_while(stmt, nextt=nextt, prefix=prefix)
         elif isinstance(stmt, pyast.For):
             cur_node = self.__parse_for(stmt, nextt=nextt, prefix=prefix)
-            logging.debug(f"For {cur_node.unique_id} {cur_node} -> {nextt.unique_id}")
+            logging.debug(
+                "For %s %s => %s", cur_node.unique_id, cur_node, nextt.unique_id
+            )
         elif isinstance(stmt, pyast.If):
             cur_node = self.__parse_ifelse(stmt=stmt, nextt=nextt, prefix=prefix)
         elif isinstance(stmt, pyast.Expr):
```

Comment:
Is there a reason why this can't be a match statement like before?

Reply:
I'm still a bit hesitant on ditching Python 3.9. I plan to only use `match` if it can avoid a lot of nested statements.

Reply:
I think it's either go big or go home, not in-between.

Since one section uses `match` then you already need to ditch Python 3.9.

Reply:
Fair enough. Go home it is.